app/controllers/admin/base_controller.rb

```diff
@@ -0,0 +1,7 @@
+class Admin::BaseController < ApplicationController 
+    before_action :require_admin
+
+    def require_admin 
+        render file: "/public/404" unless role == "Admin"
+    end
```

Comment:
I like this method to test the current role of a user to check for authorization